### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by zhaojj2209**
Changes:
- Updated developer guide with user stories for v1.2 and v1.3.
- Modified developer guide AB3 to Fine$$e.
- Modified relevant diagrams from AB3 to Fine$$e, and deleted other irrelevant AB3 documents.
- Added developer guide feature implementation description for find command.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -21,15 +21,9 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
-
-</div>
-
-**`Main`** has two classes called [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/MainApp.java). It is responsible for,
-* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
-* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+**`Main`** has two classes called [`Main`](https://github.com/AY2021S1-CS2103T-W16-3/tp/blob/master/src/main/java/ay2021s1_cs2103_w16_3/finesse/Main.java) and [`MainApp`](https://github.com/AY2021S1-CS2103T-W16-3/tp/blob/master/src/main/java/ay2021s1_cs2103_w16_3/finesse/MainApp.java). It is responsible for:
+* On app launch: Initialising the components in the correct sequence, and connecting them to each other.
```
Comment (yongping827) [inline docs/DeveloperGuide.md:25]:
```suggestion
* On app launch: Initializing the components in the correct sequence, and connecting them to each other.
```

**Review by yongping827 — APPROVED** [review]
LGTM after the suggested change.

**Review by yongping827 — COMMENTED** [review]
In `StorageClassDiagram.puml`:
* `JsonSerializableFinanceTracker` instead of `JsonSerializableFinanceTrackerStorage`
* `JsonSerializableFinanceTracker` should have lists of `JsonAdaptedExpense` and `JsonAdaptedIncome`

In `UiClassDiagram.puml`:
* Missing `ExpensePanel` and `IncomePanel`

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by siddarth2824 — APPROVED** [review]
LGTM!

**Pull request merged**